.github/workflows/test_latest_versions.yml

```diff
@@ -41,6 +41,4 @@ jobs:
         shell: bash
         run: |
           toxpyversion=$(echo ${{ matrix.python-version }} | sed -E 's/^([0-9]+)\.([0-9]+).*$/\1\2/')
-          tox -epy${toxpyversion}
-          tox -epy${toxpyversion}-notebook
-          tox -edoctest
+          tox -epy${toxpyversion}-all
```

Comment:
I am concerned that if we merge this PR, it will leave nothing in CI to enforce that the dependencies are actually specified correctly.  We could import from a notebook-only dependency in a unit test, for instance, and it would never complain.

Could we perhaps instead merge the doctest and notebook environments?  They serve a very similar purpose.  Just an idea.